compiler/native/expand_test.go

```diff
@@ -218,6 +219,77 @@ func TestNative_ExpandSteps(t *testing.T) {
 	}
 }
 
+func TestNative_ExpandStepsStarlark(t *testing.T) {
+	// setup context
+	gin.SetMode(gin.TestMode)
+
+	resp := httptest.NewRecorder()
+	_, engine := gin.CreateTestContext(resp)
+
+	// setup mock server
+	engine.GET("/api/v3/repos/foo/bar/contents/:path", func(c *gin.Context) {
+		c.Header("Content-Type", "application/json")
+		c.Status(http.StatusOK)
+		c.File("testdata/template-starlark.json")
+	})
+
+	s := httptest.NewServer(engine)
+	defer s.Close()
+
+	// setup types
+	set := flag.NewFlagSet("test", 0)
+	set.Bool("github-driver", true, "doc")
+	set.String("github-url", s.URL, "doc")
+	set.String("github-token", "", "doc")
+	c := cli.NewContext(nil, set, nil)
+
+	tmpls := map[string]*yaml.Template{
+		"go": {
+			Name:   "go",
+			Source: "github.example.com/foo/bar/template.star",
+			Format: "starlark",
+			Type:   "github",
+		},
+	}
+
+	steps := yaml.StepSlice{
+		&yaml.Step{
+			Name: "sample",
+			Template: yaml.StepTemplate{
+				Name:      "go",
+				Variables: map[string]interface{}{},
+			},
+		},
+	}
+
+	want := yaml.StepSlice{
+		&yaml.Step{
+			Commands: []string{"go build", "go test"},
+			Image:    "golang:latest",
+			Name:     "sample_build",
+			Pull:     "not_present",
+		},
+	}
+
+	// run test
+	compiler, err := New(c)
+	if err != nil {
+		t.Errorf("Creating new compiler returned err: %v", err)
+	}
+
+	got, err := compiler.ExpandSteps(steps, tmpls)
+	if err != nil {
+		t.Errorf("ExpandSteps returned err: %v", err)
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		if diff := cmp.Diff(want, got); diff != "" {
+			t.Errorf("MakeGatewayInfo() mismatch (-want +got):\n%s", diff)
```

Comment:
I think this error message is slightly off here.. But maybe I'm missing something?

Reply:
Fixed. This appears to have been a copy/paste mistake from some example.

Reply:
We can just remove the `cmp` code being used. It was used for debugging purposes

Reply:
Removed all references of it